components/director/internal/repo/upsert.go

```diff
@@ -0,0 +1,61 @@
+package repo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/lib/pq"
+
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/pkg/errors"
+)
+
+type Upserter struct {
+	tableName       string
+	columns         []string
+	conflictColumns []string
+	updateColumns   []string
+}
+
+func NewUpserter(tableName string, columns []string, conflictColumns []string, updateColumns []string) *Upserter {
```

Comment:
think about better names, my proposal is:
- createColumns
- updateColumns
- identifyingColumns or idColumns

Reply:
How about:
- createColumns
- conflictingColumns
- updateColumns